### ISSUE TITLE
"hoistPatternLet" inserts "let" keyword before "import"

### ISSUE THREAD

**Issue by fellipecaetano** [nicklockwood/SwiftFormat#182]
Enabling the `hoistPatternLet` rule causes a `let` keyword to be inserted before an `import`. Perhaps coincidentally, it's usually the only `import` in the source file. When `hoistPatternLet` is disabled the bogus `let` keyword is not inserted.

Running
```bash
swiftformat --enable hoistPatternLet
```

over the file
```swift
import PromiseKit

func HomeReducer(state: HomeState, action: Action) -> HomeState {
    switch (state, action) {
    case (.updated(let previousFeed), HomeAction.loading):
        return .loading(previous: previousFeed)
    default:
        return state
    }
}
```

produces the output
```swift
let import PromiseKit

func HomeReducer(state: HomeState, action: Action) -> HomeState {
    switch (state, action) {
    case (.updated(previousFeed), HomeAction.loading):
        return .loading(previous: previousFeed)
    default:
        return state
    }
}
```

It also happens to exactly one other source file in my codebase. If you'd like to take a look at it just let me know.

**Comment by nicklockwood** [comment]
I’ll investigate, thanks! One repro sample should be enough, unless you think the other case might be a different issue (i.e. not related to import).

**Comment by fellipecaetano** [comment]
Not at all, both files are actually very similar. Thanks!

**Comment by nicklockwood** [comment]
Fixed in 0.29.9
